src/app/components/new-goal/new-goal.component.ts

```diff
@@ -36,4 +40,14 @@ export class NewGoalComponent implements OnInit {
       this.avgWorkload = this.goalWorkload / this.diffDays;
     }
   }
+
+  onCreateGoal() {
+    this.goalManagementService.setGoal(
+        this.goalName, this.goalStartDate, this.goalEndDate, this.goalWorkload,
+        this.avgWorkload);
+
+    let goalKey = this.goalName + this.route.params['username'];
```

Comment:
As I mentioned in the other comment, here 'username' can also be referenced from the route definition file